From cfe22dff6a18102738f17e2500efc4cdc8a28be0 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Wed, 14 Nov 2018 16:20:36 +0700
Subject: [PATCH 134/174] drm: rcar-du: Fix DU1 restart condition

When the display timing and dot clock generator 1 for planes
associated with DU1 are selected, DU1 should always be restarted.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Hung Dong <hung.dong.xd@rvc.renesas.com>
Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
Signed-off-by: Nhan Nguyen <nhan.nguyen-thanh@banvien.com.vn>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index 4fcb1bb1b843..36f1c026adea 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -676,8 +676,7 @@ static void rcar_du_crtc_update_planes(struct rcar_du_crtc *rcrtc)
 				    (dptsr_planes << 16) | dptsr_planes);
 		rcrtc->group->dptsr_planes = dptsr_planes;
 
-		if (rcrtc->group->used_crtcs)
-			rcar_du_group_restart(rcrtc->group);
+		rcar_du_group_restart(rcrtc->group);
 	}
 
 	/* Restart the group if plane sources have changed. */
-- 
2.25.1

